uint16 td1
uint16 td2
uint16 td3
uint16 td4
